barretenberg/cpp/src/barretenberg/proof_system/op_queue/ecc_op_queue.hpp

```diff
@@ -29,36 +16,104 @@ struct ECCOp {
+
+    std::array<Point, 4> ultra_ops_commitments;
+    std::array<Point, 4> previous_ultra_ops_commitments;
+
+    Point get_accumulator() { return accumulator; }
+
+    /**
+     * @brief Set the current and previous size of the ultra_ops transcript
+     *
+     * @details previous_ultra_ops_size = M_{i-1} is needed by the prover to extract the previous aggregate op
+     * queue transcript T_{i-1} from the current one T_i. This method should be called when a circuit is 'finalized'.
+     */
+    void set_size_data()
+    {
+        previous_ultra_ops_size = current_ultra_ops_size;
+        current_ultra_ops_size = ultra_ops[0].size();
+    }
 
-    uint32_t get_number_of_muls()
+    [[nodiscard]] size_t get_previous_size() const { return previous_ultra_ops_size; }
+    [[nodiscard]] size_t get_current_size() const { return current_ultra_ops_size; }
+
+    void set_commitment_data(std::array<Point, 4>& commitments)
     {
-        uint32_t num_muls = 0;
-        for (auto& op : raw_ops) {
-            if (op.mul) {
-                if (op.scalar_1 != 0) {
-                    num_muls++;
-                }
-                if (op.scalar_2 != 0) {
-                    num_muls++;
-                }
-            }
+        previous_ultra_ops_commitments = ultra_ops_commitments;
+        ultra_ops_commitments = commitments;
+    }
+
+    /**
+     * @brief Get a 'view' of the current ultra ops object
+     *
+     * @return std::vector<std::span<Fr>>
+     */
+    std::vector<std::span<Fr>> get_aggregate_transcript()
+    {
+        std::vector<std::span<Fr>> result;
+        result.reserve(ultra_ops.size());
+        for (auto& entry : ultra_ops) {
+            result.emplace_back(entry);
         }
-        return num_muls;
+        return result;
     }
 
-    Point get_accumulator() { return accumulator; }
+    /**
+     * @brief Get a 'view' of the previous ultra ops object
+     *
+     * @return std::vector<std::span<Fr>>
+     */
+    std::vector<std::span<Fr>> get_previous_aggregate_transcript()
+    {
+        std::vector<std::span<Fr>> result;
+        result.reserve(ultra_ops.size());
+        // Construct T_{i-1} as a view of size M_{i-1} into T_i
+        for (auto& entry : ultra_ops) {
+            result.emplace_back(entry.begin(), previous_ultra_ops_size);
+        }
+        return result;
+    }
+
+    /**
+     * @brief TESTING PURPOSES ONLY: Populate ECC op queue with mock data as stand in for "previous circuit" in tests
+     * @details (Issue #723) We currently cannot support Goblin proofs (specifically, transcript aggregation) if there
+     * is not existing data in the ECC op queue (since this leads to zero-commitment issues). This method populates the
+     * op queue with mock data so that the prover of an arbitrary 'first' circuit can behave as if it were not the
+     * prover over the first circuit in the stack.
+     *
+     * @param op_queue
+     */
+    void populate_with_mock_initital_data()
```

Comment:
Could you create a todo to delete later?

Reply:
Added a TODO with reference to existing issue to make it clear this should eventually go away entirely